### PR TITLE
[docs] restructuring content for Indexes and Constraints.

### PR DESCRIPTION
**Pull request by aishwarya24**
Summary 
Build URL :https://deploy-preview-11374--infallible-bardeen-164bc9.netlify.app/latest/explore/indexes-constraints/

This PR addresses the following section from the issue #10923:
- **Indexes and Constraints** 
    - Primary Key (YSQL and YCQL)
    - Foreign Key (YSQL)
    - Other Constraints (YSQL)
    - Unique Index (YSQL and YCQL)
    - Partial Index (YSQL and YCQL)
    - Expression Index (YSQL)
    - GIN Index (YSQL)

The structure now looks like :
 - **Indexes and Constraints** 
    - Overview(Indexes)
    - Unique Indexes (YSQL and YCQL)
    - Partial Indexes (YSQL and YCQL)
    - Expression Indexes (YSQL)
    - GIN Index (YSQL) 
    - Primary Key (YSQL and YCQL)
    - Foreign Key (YSQL)
    - Other Constraints (YSQL)
  
- All the pages except GIN needs review for this PR. (Will update @ymahajan's [PR](https://github.com/yugabyte/yugabyte-db/pull/10759) for GIN after this one gets merged)
- The scope of the explore pages is to include simple introduction to these concepts and link to more details under Reference in docs and YB blog posts(wherever relatable)
- Some pages may have examples and some may have not have elaborate ones. This is because the reference section covers them in detail.

### CODE REVIEW

File: docs/content/latest/explore/indexes-constraints/expression-index-ysql.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    </a>
+  </li>
+</ul>
+
+Indexes usually are created based on the columns, but the expression indexes lets you create an index based on a generic expression(function or modification of data entered) involving the table columns.
+
+## Syntax
+
+```ysql
+CREATE INDEX index_name ON table_name(expression);
```
Comment (FranckPachot) [inline docs/content/latest/explore/indexes-constraints/expression-index-ysql.md:32]: I think the syntax is with parentheses around the expression:
```
 CREATE INDEX index_name ON table_name( (expression) );
```
They can be omitted in some cases (like this one with a function) but in general they should be there

File: docs/content/latest/explore/indexes-constraints/expression-index-ysql.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+  <li >
+    <a href="../expression-index-ysql/" class="nav-link active">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL
+    </a>
+  </li>
+</ul>
+
+Indexes usually are created based on the columns, but the expression indexes lets you create an index based on a generic expression(function or modification of data entered) involving the table columns.
```
Comment (FranckPachot) [inline docs/content/latest/explore/indexes-constraints/expression-index-ysql.md:27]: Should be helpful to mention that they are also called Function Based Indexes sometimes

File: docs/content/latest/explore/indexes-constraints/overview.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+*column_list* represents a column or a comma-separated list of several columns to be stored in the index. An index created for more than one column is called a composite index.
+
+You can also create a functional index in YSQL, in which case you would replace any element of *column_list* with an expression. For more information, see [Expression Indexes](../../../explore/indexes-constraints/expression-index-ysql/).
+
+YSQL currently supports index access methods `lsm` (log-structured merge-tree) and `ybgin`. These indexes are based on YugabyteDB's DocDB storage and are similar in functionality to PostgreSQL's `btree` and `gin` indexes, respectively. The index access method can be specified with `USING <access_method_name>` after *table_name*. By default, `lsm` is chosen. For more information on `ybgin`, see [Generalized inverted index](../../../explore/indexes-constraints/gin/).
+
+You can apply sort order on the indexed columns as `ASC` (default), `DESC`, as well as `HASH`. For examples, see [HASH and ASC examples in YSQL](../../../api/ysql/the-sql-language/statements/ddl_create_index/#unique-index-with-hash-column-ordering)
```
Comment (FranckPachot) [inline docs/content/latest/explore/indexes-constraints/overview.md:35]: I think that HASH is the default for the first column and HASH for the others

File: docs/content/latest/explore/indexes-constraints/primary-key-ysql.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+CREATE TABLE employees (
+  employee_no integer,
+  name text,
+  department text
+);
+
+ALTER TABLE employees ADD PRIMARY KEY (employee_no);
+```
+
+- The `ALTER TABLE` statement also allows you to add an auto-incremented primary key to an existing table by using the [SERIAL type](../../ysql-language-features/data-types/#serial-pseudotype), as per the following syntax:
```
Comment (FranckPachot) [inline docs/content/latest/explore/indexes-constraints/primary-key-ysql.md:126]: Shouldn't we recommend `bigint generated always as identity` rather `serial`? It has more possibilities, like setting the cache size.

File: docs/content/latest/explore/indexes-constraints/overview.md (hunk 3 above)
Comment (aishwarya24) [reply docs/content/latest/explore/indexes-constraints/overview.md:35]: Yes indeed! HASH for the first column and ASC for the second. I saw it here : https://docs.yugabyte.com/latest/api/ysql/the-sql-language/statements/ddl_create_index/#expression
Thank you. I'll change accordingly.

**Review by polarweasel — APPROVED** [review]
LGTM

**Pull request merged**